x/fibre/blob.go

```diff
@@ -0,0 +1,307 @@
+package fibre
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"runtime"
+
+	"github.com/celestiaorg/rsema1d"
+	"github.com/celestiaorg/rsema1d/field"
+)
+
+// ErrBlobTooLarge is returned when the blob size exceeds MaxBlobSize.
+var ErrBlobTooLarge = errors.New("blob size exceeds maximum allowed size")
```

Comment:
[nit] can this include Fibre to differentiate it from the existing blobs?

```suggestion
// ErrFibreBlobTooLarge is returned when the Fibre blob size exceeds MaxFibreBlobSize.
var ErrFibreBlobTooLarge = errors.New("fibre blob size exceeds maximum allowed size")
```

Reply:
The actual name of this var is `fibre.ErrBlobTooLarge`. Basically everything in this pkg is already domain separated, so I don't see much reason to duplicate the separator

Reply:
Fair enough, the type is in the `fibre` package so it doesn't need it. 

The error message may still warrant a "fibre" because we already have these:

https://github.com/celestiaorg/celestia-app/blob/e93e3c24e729e734381266b15fd36b87cfd48352/x/blob/types/errors.go#L37-L38

but I'll defer to you.

Reply:
agree on the error string

Reply:
I added a change in the client, which prefixes any returned error with "fibre: ", even if it's not predefined one